### PR TITLE
BAU: Allow DatabaseConfiguration to look at VCAP_SERVICES

### PR DESCRIPTION
**Pull request by tlwr**
What
----

Instead of passing `url` in as an environment variable we can look at the
VCAP_SERVICES environment variable provided by Cloud Foundry.

Why
---

We have this script:

https://github.com/alphagov/verify-stub-idp/blob/master/deploy_to_paas.sh

Which could be replaced by:

```
$ cat manifest.yml
---
memory: 1G
buildpack: java_buildpack
env:
	LOG_PATH: /app/stub-idp/logs
	LOG_LEVEL: INFO
	CONFIG_FILE: /app/stub-idp/stub-idp.yml
	ENTITY_ID: "http://stub_idp.acme.org/%s/SSO/POST"
	KEY_TYPE: encoded
	CERT_TYPE: encoded
	TRUSTSTORE_TYPE: encoded
	TRUSTSTORE_PASSWORD: puppet
	STUB_IDP_BASIC_AUTH: true
	GRAPHITE_REPORTING_FREQUENCY: 100days
	STUB_IDP_SIGNING_PRIVATE_KEY ((stub_idp_signing_private_key))
	STUB_IDP_SIGNING_CERT ((stub_idp_signing_cert))
	METADATA_TRUSTSTORE ((metadata_truststore))
	STUB_IDPS_FILE_PATH /app/stub-idp/resources/((environment))/stub-idps.yml
	METADATA_URL ((paas_metadata_url))
	METADATA_ENTITY_ID ((metadata_entity_id))
	EUROPEAN_IDENTITY_ENABLED ((european_identity_enabled))
	SINGLE_IDP_FEATURE ((single_idp_feature))
	SERVICE_LIST_URL ((service_list_url))
	VERIFY_SUBMISSION_URL ((verify_submission_url))
	STUB_COUNTRY_SIGNING_PRIVATE_KEY ((stub_country_signing_private_key))
	STUB_COUNTRY_SIGNING_CERT ((stub_country_signing_cert))
	STUB_IDP_HOSTNAME https://((hostname)).((cf_domain))
	HUB_CONNECTOR_ENTITY_ID ((hub_connector_entity_id))
services:
- ida-stub-idp-db
- ida-stub-idp-logit
...
```

and

```
$ cf v3-zdt-push stub-idp-new --vars-file /path/to/vars/file
```

TLDR; make it simpler

Who can review
--------------

Not @tlwr

### CODE REVIEW

File: stub-idp/src/test/java/uk/gov/ida/stub/idp/configuration/BaseConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.ida.stub.idp.configuration;
+
+import io.dropwizard.configuration.ConfigurationSourceProvider;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class BaseConfigurationTest {
```
[the remark this answers is not in the document]
Comment (tlwr) [reply stub-idp/src/test/java/uk/gov/ida/stub/idp/configuration/BaseConfigurationTest.java]: Yes, I guess this could be `ConfigurationTestHelper` or something
basically I wanted to reuse the code from `StubIdpConfigurationTest` in `DatabaseConfigurationTest`.
